Library/MeterGeometry.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+#ifndef __METERGEOMETRY_H__
+#define __METERGEOMETRY_H__
+
+#include "Meter.h"
+#include <vector>
+#include "..\Common\Gfx\Shape.h"
+
+class MeterGeometry :
+	public Meter
+{
+public:
+	MeterGeometry(Skin* skin, const WCHAR* name);
+	~MeterGeometry();
+
+	MeterGeometry(const MeterGeometry& other) = delete;
+	MeterGeometry& operator=(MeterGeometry other) = delete;
+
+	virtual UINT GetTypeID() { return TypeID<MeterGeometry>(); }
+
+	virtual void Initialize();
+	virtual bool Update();
+	virtual bool Draw(Gfx::Canvas& canvas);
+
+protected:
+	virtual void ReadOptions(ConfigParser& parser, const WCHAR* section);
+	virtual void BindMeasures(ConfigParser& parser, const WCHAR* section);
+
+	//virtual bool IsFixedSize(bool overwrite = false) { return overwrite ? true : m_ImageName.empty(); }
+
+private:
+	struct GeometryShape : Gfx::Shape {
```

Comment:
It could, but i want to make Gfx::Shape only contain the fields needed to actually render the shape, and let all possible calculations (rotation, offset, skew, etc) happen in the Meter class instead of in the DrawGeometry method.